### PR TITLE
Fix metamask connection cancel

### PR DESCRIPTION
**Pull request by dominikrudzki**
fixes https://github.com/ethereum/remix-project/issues/5293

Environment won't change until we accept the metamask connection

### CODE REVIEW

File: apps/remix-ide/src/app/providers/injected-provider.tsx [hunk 1]
```diff
@@ -61,7 +65,12 @@ export abstract class InjectedProvider extends Plugin implements IProvider {
       this.call('notification', 'toast', this.notFound())
       throw new Error(this.notFound())
     } else {
-      this.askPermission(true)
+      try {
+        await this.askPermission(true)
+      } catch (error) {
+        this.call('notification', 'toast', this.notFound())
```
Comment (LianaHus) [inline apps/remix-ide/src/app/providers/injected-provider.tsx:71]: The message does not say anything about the connection.
We can prepend the text here.
Something like " Please make sure your Injected Provider is connected to Remix." @Aniket-Engg ?

Comment (Aniket-Engg) [reply apps/remix-ide/src/app/providers/injected-provider.tsx:71]: looks fine to me

Comment (dominikrudzki) [reply apps/remix-ide/src/app/providers/injected-provider.tsx]: Thanks for the suggestion. I've made the change in the latest commit. Let me know if there's anything else to tweak :)

**Review by LianaHus — APPROVED** [review]

**Comment by LianaHus** [comment]
LGTM @Aniket-Engg . Feel free to merge.
@dominikrudzki thanks for the PR!

**Pull request merged**